### PR TITLE
Eq and taxon bugs

### PR DESCRIPTION
**Pull request by kshefchek**
Small changes to help with equivalence and category issues:
- NCBI gene: make ensembls xrefs only (no equivalent triples)
- MGI: remove sameAs between marker and alleles (see #519)
- GO: remove interacting taxon as taxon for the gene itself

### CODE REVIEW

File: dipper/sources/GeneOntology.py [hunk 1]
```diff
@@ -371,8 +371,10 @@ def process_gaf(self, gaffile, limit, id_map=None):
                         else:
                             model.addSynonym(gene_id, syn)
 
-                for txid in taxon.split('|'):
-                    tax_curie = re.sub(r'taxon:', 'NCBITaxon:', txid)
+                # First taxon is for the gene, after the pipe are interacting taxa
+                tax_curie = taxon.split('|')[0].replace('taxon', 'NCBITaxon')
+                # this is a required field but good to safe
+                if tax_curie:
                     geno.addTaxon(tax_curie, gene_id)
```
Comment (TomConlin) [inline dipper/sources/GeneOntology.py:378]: might be interesting to check if exists and if so,  is it different?
also might be unnecessarily slow

Comment (kshefchek) [reply dipper/sources/GeneOntology.py:378]: if it's different from the expected taxon for the file?

File: dipper/sources/NCBIGene.py [hunk 2]
```diff
@@ -470,7 +473,8 @@ def _add_gene_equivalencies(self, dbxrefs, gene_id, taxon):
                         # and ncbi is never a human clique leader in any case
                         dbxref_curie = None
                         continue
-                # designate clique leaders
+
+                # designate clique leaders and equivalentClass/sameAs triples
```
Comment (TomConlin) [inline dipper/sources/NCBIGene.py:477]: based on  dict dipper/resources/clique_leaders.yaml

**Comment by TomConlin** [comment]
+1

**Review by TomConlin — APPROVED** [review]
detecting discrepancies within a file  which most likely is not worth looking for unless we notice is being an actual problem

**Pull request merged**